### PR TITLE
[SPARK-10577] [PySpark] DataFrame hint for broadcast join

### PR DESCRIPTION
**Pull request by snowmoon-zhang**
https://issues.apache.org/jira/browse/SPARK-10577

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -189,6 +190,14 @@ def approxCountDistinct(col, rsd=None):
     return Column(jc)
 
 
+@since(1.6)
+def broadcast(df):
+    """Marks a DataFrame as small enough for use in broadcast joins."""
```
Comment (rxin) [inline python/pyspark/sql/functions.py]: can you add a test for this?

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -189,6 +190,14 @@ def approxCountDistinct(col, rsd=None):
...
+
+    sc = SparkContext._active_spark_context
+    return DataFrame(sc._jvm.functions.broadcast(df._jdf),sc._jsc)
```
Comment (rxin) [inline python/pyspark/sql/functions.py]: add space after comma

File: python/pyspark/sql/functions.py (hunk 1 above)
Comment (Jianfeng-chs) [inline python/pyspark/sql/functions.py]: Reynold Xin, I have added one test for this.

**Comment by Jianfeng-chs** [comment]
Since messed commits, closing it

**Pull request closed without merging**